docker/pusher.go

```diff
@@ -23,46 +29,66 @@ func NewPusher() *Pusher {
 }
 
 // Push the image of the Function.
-func (n *Pusher) Push(f bosonFunc.Function) (digest string, err error) {
-	// Check for the docker binary explicitly so that we can return
-	// an extra-friendly error message.
-	_, err = exec.LookPath("docker")
-	if err != nil {
-		err = errors.New("please install 'docker'")
-		return
-	}
+func (n *Pusher) Push(ctx context.Context, f bosonFunc.Function) (digest string, err error) {
 
 	if f.Image == "" {
 		return "", errors.New("Function has no associated image.  Has it been built?")
 	}
 
-	// set up the command, specifying a sanitized project name and connecting
-	// standard output and error.
-	cmd := exec.Command("docker", "push", f.Image)
+	parts := strings.Split(f.Image, "/")
+	if len(parts) < 1 {
+		return "", errors.Errorf("invalid image name: %q", f.Image)
+	}
+	registry := parts[0]
```

Comment:
Currently, because the docker CLI supports it, we accepted just the namespace for the docker.io registry. For example `lanceball` instead of `docker.io/lanceball`. With this change, does that default to docker.io still work? At first glance, it doesn't seem like it would, because the value of `registry` here has to match what's in `~/.docker/config.yaml`.

Reply:
Fixed: if the `namespace/image:tag` format is used then `docker.io` registry is assumed.